File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 10/08/2026, 02:37:38
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map;
        int n = nums.size();
        vector<int> ans;
        for(int i = 0; i<n; i++){
            int rem = target-nums[i];
            if(map.find(rem) != map.end()){
                ans.push_back(map[rem]);
                ans.push_back(i);
            }   
            else map[nums[i]] = i;
        }
        return ans;
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 10/08/2026, 02:37:33
class Solution(object):
    def twoSum(self, nums, target):
        seen = {}
        for i, num in enumerate(nums):
            if (target - num) in seen:
                return [seen[target - num], i]
            else:
                seen[num] = i
        